packages/nc-gui/components/account/Token.vue

```diff
@@ -75,6 +94,9 @@ const deleteToken = async (token: string): Promise<void> => {
 }
 
 const generateToken = async () => {
+  isValidTokenName.value = !selectedTokenData.value.description?.length || selectedTokenData.value.description?.length > 255
```

Comment:
since `isValidTokenName` is affirmative, your logic should be inverted now. Please check all the occurrences.

Reply:
its working  fine now

Reply:
ya got it. , i have renamed the varaible to isNotValidTokenName , i tried changing the logic to opposite it is throwing some ts error so I changed the variable name

Reply:
What error you got? I don't think this is an acceptable reason.

Reply:
```
  isValidTokenName.value = selectedTokenData.value.description?.length && selectedTokenData.value.description?.length <= 255

```

![image](https://github.com/nocodb/nocodb/assets/66205494/b2d7ded7-e723-45d9-9acb-2f13260b2872)

Reply:
i have changed it , added a helper function , musthfa helped me, I was bit confused 😞

Reply:
by using `?.`, it could be possibly `undefined`, in this case this value cannot assign to a boolean type. You can  
simply check if `selectedTokenData.value.description` is undefined first.

Reply:
also i think if you reset `selectedTokenData.value` with default value, then `?.` is not necessary at all cause it must have description in this case.

Reply:
i have moved the code to like this

```typescript
const validateTokenName = (tokenName: string | undefined) => {
  if (!tokenName) return false
  return tokenName.length < 255
}
```

Reply:
```typescript
 isValidTokenName.value = validateTokenName(selectedTokenData.value.description)
```

Reply:
Good. Resolved this convo.